lxd/instance/drivers/qmp/commands.go

```diff
@@ -406,6 +404,23 @@ func (m *Monitor) RemoveBlockDevice(blockDevName string) error {
 	return nil
 }
 
+// AddDevice adds a new device.
+func (m *Monitor) AddDevice(device map[string]interface{}) error {
```

Comment:
Why does this need `map[string]interface{}`?

Reply:
It doesn't need to be `map[string]interface{}` but I changed it for two reasons:
1) The value can be something other than a string. We only use strings though.
2) `blockdev` is a `map[string]interface{}` and I wanted to keep things consistent.

Reply:
If it doesn't need to be changed just yet the lets not, see https://github.com/canonical/lxd-cloud/pull/65#discussion_r837821089 on thoughts around using the most restrictive data type possible.

And in the future if we need to call `AddDevice()` with something other than `map[string]string` then we can change the `AddDevice` signature but then we would still not need to change the signatures of the existing callers of `AddDevice` unless they changed their usage too. 

Ideally if we do get to a point where `AddDevice` needs to accept `map[string]interface{}` it would likely be better to then create an interface called `Device` that has a `Marshal` function into the qmp protocol, and then implement a struct per-device-type that can be passed to it (with the per-device fields). That way we continue to benefit from Go's static typing and the possibilities of what to provide to the function are self-documenting.